src/pudl/analysis/allocate_net_gen.py

```diff
@@ -21,46 +21,83 @@
   about how much generation and fuel consumption is attributable to an entire
   plant.
 * The generators_eia860 table provides an exhaustive list of all generators
   whose generation is being reported in the generation_fuel_eia923 table.
 
-We allocate the net generation reported in the generation_fuel_eia923 table on the basis
-of plant, prime mover, and fuel type among the generators in each plant that have
-matching fuel types. Generation is allocated proportional to reported generation if it's
-available, and proportional to each generator's capacity if generation is not available.
-
-In more detail: within each year of data, we split the plants into three groups:
-
-* Plants where ALL generators report in the more granular generation_eia923
-  table.
-* Plants where NONE of the generators report in the generation_eia923 table.
-* Plants where only SOME of the generators report in the generation_eia923
-  table.
-
-In plant-years where ALL generators report more granular generation, the total net
-generation reported in the generation_fuel_eia923 table is allocated in proportion to
-the generation each generator reported in the generation_eia923 table. We do this
-instead of using net_generation_mwh from generation_eia923 because there are some small
-discrepancies between the total amounts of generation reported in these two tables.
-
-In plant-years where NONE of the generators report more granular generation, we create a
-generator record for each associated fuel type. Those records are merged with the
+There are six main stages of the allocation process in this module:
+
+* Inputs! Grabbing the input tables from :class:`pudl.output.pudl.PudlTabl`. Some
+  cleaning (like :func:`distribute_annually_reported_data_to_months_if_annual`) and
+  aggregation (like :func:`allocate_bf_data_to_gens`) happen in this stage.
+* Associating the data from the input tables with generator/prime mover/energy source
+  codes - using the primary keys in :py:const:`IDX_GENS_PM_ESC` (see
+  :func:`associate_generator_tables`).
+* Prepare for the allocation (see :func:`prep_alloction_fraction`) by adding boolean
+  flags for whether a particular :py:const:`IDX_GENS_PM_ESC` shows up in each of the
+  input data tables.
+* Allocation! Do the actual allocating. See :func:`allocate_net_gen_by_gen_esc` for net
+  generation allocation and :func:`allocate_fuel_by_gen_esc` for fuel allocation. More
+  details on the allocation process below.
+* Tests! The high-level test of whether the allocated data matches the original inputs
+  is :func:`test_the_original_gf_vs_the_allocated_by_gens_gf`. There are also other
+  warnings that are logged if assumptions aren't met via :func:`_test_frac`,
+  :func:`_test_gen_fuel_allocation` and :func:`_test_gen_pm_fuel_output`
+* Aggregate the main output with primary keys of :py:const:`IDX_GENS_PM_ESC` to the
+  generator-level with primary keys of :py:const:`IDX_GENS` via
+  :func:`aggregate_gen_fuel_by_generator`.
```

Comment:
* Use the same form for each of these bullets -- imperative mood (e.g. "Aggregate outputs to generator level" not "Aggregating outputs to generator level.") and the same format (e.g. start with a 1-2 word name for the step, maybe in bold, if we want to be able to refer back to them later).
* "Some cleaning" in Inputs is vague. What kind of processing has to happen in this stage rather than later? Can the gist of it (and how it's different from other steps) be conveyed in a line or two?
* The meaning of "Associate" isn't yet clear at this point in the document. This would be the place to first define it.
* Referring to `IDX_GENS_PM_ESC` here multiple times without also explicitly saying what it means is confusing. I think it's the primary keys of the `generation_fuel` table, right?
* Add at least a one-line description of what's being done, alongside the links to the functions that are doing it. "Allocation! Do the actual allocating. ... More details below." doesn't provide any information to the reader beyond the single word "Allocation". What's being allocated, and on what basis (generally speaking) and how is that distinct from the distribution and allocation that was referenced in the first bullet point?
* "Prepare for the allocation" isn't informative -- everything to this point is preparation for the allocation. Describe what the boolean flags are and why they're needed.
* Explicitly say what it means for for allocated data to "match" the original inputs.

## Examples.
* **Read inputs:** Read denormalized net generation and fuel consumption data from the PUDL DB. Standardize reporting frequency. Allocate boiler fuel consumption to their associated generators. See (functions).
* **Sanity check allocation:** Verify that the total allocated net generation and fuel consumption within each plant is equal to the total of the originally reported values (within some tolerance). See (functions)

Reply:
ty for these suggestions! i except you'll still want to make some additional suggestion